packages/csharp/src/name-policy.ts

```diff
@@ -0,0 +1,37 @@
+import * as core from "@alloy-js/core";
+import * as changecase from "change-case";
+
+// the context in which the name policy should be applied
+export type CSharpElements =
+  | "class"
+  | "constant"
+  | "enum"
+  | "enum-member"
+  | "function"
+  | "interface"
+  | "member"
```

Comment:
class-member might be more descriptive? Unsure of common C# naming conventions so def follow those.